python/ray/autoscaler/_private/gcp/node.py

```diff
@@ -371,12 +381,17 @@ def list_instances(
             key=TAG_RAY_CLUSTER_NAME, value=self.cluster_name
         )
 
+        # TPU VMs spawn accompanying Compute Instances that must be filtered out,
+        # else this results in duplicated nodes.
+        tpu_negation_filter_expr = "(NOT labels.{label}:*)".format(label="tpu_cores")
```

Comment:
Ideally we would have a unit test for this

Reply:
Great point, added a test in `test_gcp_node_provider.py`